### PR TITLE
Added new Shunk with updated appearance, actuated chuck and action fo…

### PR DESCRIPTION
**Pull request by RDaneelOlivav**
…r open and close and start operation and end operation. Also decoupled shunk controllers from action.
This is an update for the shunk machine. In theory, to use it, the action server has to be called for opening and closing the chuck. And also for staring the operation which won't allow opening the chuck and stopping the operation once it has started.
Also, added models with simplified collisions to maximize the grasping ability of the parts, although not tested, because thats another step.

### CODE REVIEW

**Comment by moriarty** [comment]
I will just close and re-open this to trigger the Build.

File: fetchit_challenge/action/SchunkMachine.action [hunk 1]
```diff
@@ -1,5 +1,7 @@
 uint32 OPEN=0
 uint32 CLOSE=1
+uint32 START_OPERATION=2
```
Comment (moriarty) [inline fetchit_challenge/action/SchunkMachine.action:3]: @velveteenrobot can you review this

File: fetchit_challenge/nodes/shunk_machine_controlers.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def close_chuck(self):
+        """Set Pi pins to close chuck."""
+        GPIO.output(40, 1)
+        GPIO.output(37, 0)
+
+    def start_operation(self):
+        pass
+
+    def end_operation(self):
+        pass
```
Comment (moriarty) [inline fetchit_challenge/nodes/shunk_machine_controlers.py:46]: The `.pyc` file shouldn't be checked in. it's the next file in the PR

**Review by moriarty — CHANGES_REQUESTED** [review]
1. There is a `.pyc` file which shouldn't be checked in.
2. Why did the `cmake_minimum_required` version change?
3. Is there now an additional action which must be called? START_OPERATION and END_OPERATION ? @velveteenrobot should take a look and see if this is really required we were going to just start the operation timer as soon as "close" was called.

**Comment by ben-fido** [comment]
This is very good @RDaneelOlivav , it works now.  
Thanks!

.

### Important things to fix:

**Add this to package.xml**
```
<exec_depend>joint_state_controller</exec_depend>
<exec_depend>position_controllers</exec_depend>
```

.

**Edit this file:**  
`schunk_machine_server.py`  
This node is already started under namespace `"schunk_machine"`, so the server path should be `"chuck"`
like this:  
```
self.server = actionlib.SimpleActionServer('chuck', SchunkMachineAction, self.callback, False)
```

.

**Add a new launch file:**  
`shunk_machine_start_server_real.launch`
containing this code:  
```
    <group ns="schunk_machine">
        <arg name="shunk_version" />
        <node name="shunk_machine_server" pkg="fetchit_challenge" type="schunk_machine_server.py"
              respawn="false" output="screen" />
    </group>
```

.

**Change axis origin of the door visual and collision mesh:**  
The axis origin of the sliding door is down near the floor.
Can you edit the mesh file so the axis is aligned with one edge of the part?
Then change the link offset in URDF file.
This will make it easier to fix the inertia values for physics simulation, so the door moves better.

.

Then @velveteenrobot  needs to edit her sample scripts:  
```
open_schunk_machine.py
close_schunk_machine.py
```
to send action messages to `"/schunk_machine/chuck"`

.

### Minor suggestions:

**Schunk logo** disappeared from door.

.

Rename joints:  
```
lathe_gripper1_joint
lathe_gripper2_joint
lathe_gripper3_joint
```
to be:  
```
chuck_joint1
chuck_joint2
chuck_joint3
```
because there is one thing with 3 joints.

.

**Comment by moriarty** [comment]
@velveteenrobot I'm going to merge this.

The Schunk Machine is being packed today I don't know if we'll get a chance to test it but we'll have one day in Montreal before the teams arrive.

**Review by moriarty — APPROVED** [review]

**Pull request merged**